server/api/shared.py

```diff
@@ -92,6 +92,31 @@ def batch_tallies(contest: Contest) -> BatchTallies:
     }
 
 
+class CombinedBatch(TypedDict):
+    name: str
+    representative_batch: Batch
+    sub_batches: List[Batch]
+
+
+def combined_batch_representative(sub_batches: List[Batch]) -> Batch:
+    assert len(sub_batches) > 0
```

Comment:
Any additional need to assert that the sub_batches are unique?

Reply:
It's not something I'm super worried about given the context (since we're never dealing with lists of batches that have duplicates)